userspace/plugin/plugin_api.h

```diff
@@ -970,13 +1029,36 @@ typedef struct
 	//
 	// Return an updated set of metrics provided by this plugin.
 	// Required: no
+	// Arguments:
 	// - s: the plugin state, returned by init(). Can be NULL.
 	// - num_metrics: lenght of the returned metrics array.
 	//
 	// Return value: Pointer to the first element of the metrics array.
 	// 'num_metrics' must be set to the lenght of the array before returning
 	// and it can be set to 0 if no metrics are provided.
 	ss_plugin_metric* (*get_metrics)(ss_plugin_t* s, uint32_t* num_metrics);
+
+	//Capture listening capability API
+	struct
+	{
+		//
+		// Called by the framework when the event capture opens.
+		//
+		// Required: no
+		// Arguments:
+		// - s: the plugin state, returned by init(). Can be NULL.
+		// - i: input containing vtables for performing table operations and subscribe/unsubscribe async routines
+		void (*capture_open)(ss_plugin_t* s, const ss_plugin_capture_listen_input* i);
```

Comment:
Maybe make this return `ss_plugin_rc` to make it more future-proof? The same goes for `capture_close`